### PR TITLE
some bug fixes and improvements

### PR DESCRIPTION
**Pull request by ChaosEternal**

### CODE REVIEW

File: swagger_py_codegen/base.py [hunk 1]
```diff
@@ -55,7 +55,7 @@ def __init__(self):
         self.env = Environment(loader=self.loader)
 
     def add_searchpath(self, path):
-        self.loader.searchpath.append(path)
+        self.loader.searchpath.insert(0, path)
```
Comment (Rejown) [inline swagger_py_codegen/base.py]: Looks good, thank you.

File: swagger_py_codegen/templates/flask/validators.tpl [hunk 2]
```diff
@@ -85,7 +85,7 @@ def request_validate(view):
             validator = FlaskValidatorAdaptor(schema)
             result, errors = validator.validate(value)
             if errors:
-                abort(422, message='Unprocessable Entity', errors=errors)
+                abort(400, message='Invalid Request', errors=errors)
```
Comment (Rejown) [inline swagger_py_codegen/templates/flask/validators.tpl]: Why `400` is better? Will you give some reason for it?
About the `422` status code:
http://www.bennadel.com/blog/2434-http-status-codes-for-invalid-data-400-vs-422.htm
https://developer.github.com/v3/#client-errors

Perhaps left the error handling for the user is a better choice.
We will improve it.

File: swagger_py_codegen/templates/flask/validators.tpl [hunk 3]
```diff
@@ -98,8 +98,8 @@ def response_filter(view):
     def wrapper(*args, **kwargs):
         resp = view(*args, **kwargs)
 
-        if isinstance(resp, current_app.response_class):
-            return resp
+        if isinstance(resp[0], current_app.response_class):
```
Comment (Rejown) [inline swagger_py_codegen/templates/flask/validators.tpl]: `resp` can be a tuple, but also can be a response object. Reference index 0 without type checking maybe raise  `TypeError`.

File: swagger_py_codegen/base.py (hunk 1 above)
Comment (Rejown) [inline swagger_py_codegen/base.py]: Will you submit a PR about `template searchpath priority` separately? so we can merge this fleetly.

File: swagger_py_codegen/templates/flask/validators.tpl (hunk 2 above)
Comment (ChaosEternal) [inline swagger_py_codegen/templates/flask/validators.tpl]: in my case, status code 422 will trigger another error, which is weird. 

i will do more test upon this.

File: swagger_py_codegen/templates/flask/validators.tpl (hunk 3 above)
Comment (ChaosEternal) [inline swagger_py_codegen/templates/flask/validators.tpl]: maybe i have misunderstood the mechanism, will do more test.

**Pull request closed without merging**